types/react-onclickoutside/index.d.ts

```diff
@@ -1,29 +1,43 @@
 // Type definitions for react-onclickoutside v5.7.0
 // Project: https://github.com/Pomax/react-onclickoutside
-// Definitions by: Karol Janyst <https://github.com/LKay>
+// Definitions by: Karol Janyst <https://github.com/LKay> and cwmoo740 <https://github.com/cwmoo740>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 2.1
 
-import * as React from "react";
+import * as React from 'react';
 
-declare namespace ReactOnClickOutside {
-    interface OnClickOutsideComponent {
-        handleClickOutside(e: React.MouseEvent<any>): void
+declare namespace ClickOut {
+    interface HandleClickOutside<T> {
+        handleClickOutside: React.MouseEventHandler<T>;
     }
 
-    interface OnClickOutsideProps {
-        disableOnClickOutside?: boolean | Function
-        enableOnClickOutside?: Function
-        eventTypes?: string | Array<string>
-        outsideClickIgnoreClass?: string
-        preventDefault?: boolean
-        stopPropagation?: boolean
+    interface InjectedProps {
```

Comment:
I'm not sure what you mean by "none of the properties that didn't exist in the original component are actually injected."

```
function TestStateless(props: {}) {
    console.log(props);
    return (
        <div>Hello</div>
    );
}
const ClickStateless = ClickOut<{}>(TestStateless);
render(<ClickStateless/>, target);
// console.logs {enableOnClickOutside: function, disableOnClickOutside: function};
render(<ClickStateless disableOnClickOutside/>, target);
// still console.logs {enableOnClickOutside: function, disableOnClickOutside: function};
```

While testing this I did realize that eventTypes, outsideClickIgnoreClass, etc. *are* passed down to the wrapped component however, so I'll fix that. It's still the case that the enable and disable functions are "injected".

Also the other props aren't all marked with ? because it makes it easier to access those types from consuming code. It's easy to mark all fields as optional using Partial<T> but impossible to strip out an undefined from a field. Everywhere that interface is used later uses Partial, so they're still optional.